addons/crm/static/src/js/forecast/forecast_models.js

```diff
@@ -0,0 +1,163 @@
+/** @odoo-module alias=crm.ForecastModels **/
+import KanbanModel from 'web.KanbanModel';
+
+/**
+ * Would be great if we could merge this with the `web` kanban_model
+ */
+var ForecastKanbanModel = KanbanModel.extend({
+    /**
+     * Check whether to apply the forecast logic to a load or a reload,
+     * depending on the groupby and the forecast_field
+     * 
+     * Sets the current granularity if the forecast applies
+     * 
+     * @param {Object} element what to load (params) / reload (localData from id)
+     * @param {Object} params optional reload params
+     * @returns {boolean} 
+     */
+    _isForecast(element, params={}) {
+        var groupby, granularity;
+        if (typeof element === 'undefined') {
+            return false;
+        }
+        if ("groupBy" in params) {
+            [groupby, granularity] = params.groupBy[0].split(':');
+        } else if (element.groupedBy.length != 0) {
+            [groupby, granularity] = element.groupedBy[0].split(':');
+        } else if (element.groupBy.length != 0) {
+            [groupby, granularity] = element.groupBy[0].split(':');
+        }
+        if (typeof granularity === 'undefined') {
+            granularity = "month";
+        }
+        if (typeof this.forecast_field === 'undefined' || groupby != this.forecast_field) {
+            return false;
+        }
+        this.granularity = granularity;
+        return true;
+    },
+    /**
+     * Applies the forecast logic to the domain and context if needed
+     * before the read_group.
+     * 
+     * After the read_group, checks the end date of the last displayed group
+     * in order to be able to add the next group with the ForecastColumnQuickCreate widget
+     * @override
+     */
+    async __load(params) {
+        this.baseDomain = params.domain;
+        this.baseContext = params.context;
+        this.forecast_field = params.context.forecast_field;
+        if (!this._isForecast(params)) {
+            return this._super(...arguments);
+        }
+        this.min_groups =
+            params.context.fill_temporal && params.context.fill_temporal.min_groups || undefined;
+        var forecast = this.call('forecast', 'getPeriod', {
```

Comment:
`forecastService` maybe ? not that if we rename the forecast service into something else, those variables will have to be updated accordingly :D (aka "I hate Tde")

Reply:
Yep, because this.call('forecast') is quite generic when reading it, we don't have much clues about what it does. Note that if you rename it to FillTemporal something, would makes sense to rename it here also.